internal/test/e2e/basic_test.go

```diff
@@ -564,6 +564,89 @@ func TestSNISync(t *testing.T) {
 	})
 }
 
+func TestVaultSync(t *testing.T) {
+	// ensure that Vaults can be synced to Kong gateway
+	cleanup := run.Koko(t)
+
+	defer cleanup()
+
+	vault := &v1.Vault{
+		Id:     uuid.NewString(),
+		Name:   "env",
+		Prefix: "test-vault-1",
+	}
+	c := httpexpect.New(t, "http://localhost:3000")
+	c.POST("/v1/vaults").WithJSON(vault).Expect().Status(http.StatusCreated)
+
+	dpCleanup := run.KongDP(kong.GetKongConfForShared())
+	defer dpCleanup()
+
+	require.NoError(t, util.WaitForKongPort(t, 8001))
+
+	// kongClient
+	kongClient.RunWhenKong(t, ">= 3.0.0")
+
+	expectedConfig := &v1.TestingConfig{
+		Vaults: []*v1.Vault{vault},
+	}
+	util.WaitFunc(t, func() error {
+		err := util.EnsureConfig(expectedConfig)
```

Comment:
Once you do [this](https://github.com/Kong/koko/pull/464/files#r1000443742), `EnsureConfig` is no longer usable.
You'd need to create something that accepts Kong structures:

```
func EnsureKongConfig(expectedConfig KongConfig) error {
	gotConfig, err := fetchKongConfig()
	if err != nil {
		return fmt.Errorf("fetching kong config: %v", err)
	}
	err = JSONSubset(expectedConfig, gotConfig)
	return err
}
```